lib/reservation_mgr.rb

```diff
@@ -0,0 +1,111 @@
+require 'pry'
+require 'date'
+require_relative 'room'
+require_relative 'reservation'
+
+class ReservationMgr
+  attr_reader :rooms , :reservations
+  def initialize(num_rooms)
+    @rooms = Room.create(num_rooms)
+    @reservations = []
+
+  end
+
+  def available_rooms(check_in,check_out,rooms: 1)
+    if check_out < check_in
+      raise StandardError.new('The check-out date is before the check-in')
+    end
+
+    available_rooms = []
+    @rooms.each do |room|
+      if room.is_available?(check_in,check_out)
+        available_rooms << room
+      end
+    end
+
+    if available_rooms.length == 0 || available_rooms.length < rooms
+      raise ArgumentError.new('There are not enough room/rooms available')
+    end
+
+    return available_rooms
+  end
+
+
+  def make_reservation(check_in,check_out,rooms: 1, block_id: nil)
+    if block_id == nil
+      new_reservations = create_reservation(check_in,check_out,rooms, block_id)
+      return new_reservations
+
+    else
+      block_rooms = find_available_block_rooms(block_id)
+
+      if check_in != block_rooms[0].check_in || check_out != block_rooms[0].check_out
+        raise ArgumentError.new('The dates do not match for this given block reservation')
+
+      end
+
+      rooms.times do |index|
```

Comment:
You should also check to see if `block_rooms` has at least `room` number of rooms available in the block.